ci-operator/step-registry/aws/c2s/init-token-service/aws-c2s-init-token-service-commands.sh

```
@@ -77,17 +79,28 @@ temp_cred_file=$(mktemp)
+    http_code=$(curl -sS -o "${temp_cred_file}" -w "%{http_code}" "${temp_cred_request_url}" \
             --cert "${SHARED_DIR}/temp_cred_provider_cert.pem" \
             --cacert "${shift_ca_file}" \
-            --key "${SHARED_DIR}/temp_cred_provider_private_key.pem" > "${temp_cred_file}"
+            --key "${SHARED_DIR}/temp_cred_provider_private_key.pem")
+    curl_rc=$?
 
-    key_id=$(cat ${temp_cred_file}  | jq -j .Credentials.AccessKeyId)
-    key_sec=$(cat ${temp_cred_file}  | jq -j .Credentials.SecretAccessKey)
+    # Suppress jq stderr here: a non-JSON error body would otherwise emit a
+    # cryptic "parse error" that obscures the diagnostics logged below.
+    key_id=$(cat ${temp_cred_file}  | jq -j .Credentials.AccessKeyId 2>/dev/null)
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)